File: books/book6_Python核心编程（第3版）/网络编程/udp_client.py
#!/usr/bin/env python
# -*- coding: UTF-8 -*-
# coding=utf-8 

"""
@author: Li Tian
@contact: litian_cup@163.com
@software: pycharm
@file: udp_client.py
@time: 2020/12/26 15:28
@desc: udp客户端
"""

from socket import *

HOST = 'localhost'
PORT = 21567
BUFSIZ = 1024
ADDR = (HOST, PORT)

udpCliSock = socket(AF_INET, SOCK_DGRAM)

while True:
    data = input('> ')
    if not data:
        break
    udpCliSock.sendto(bytes(data, encoding="utf-8"), ADDR)
    data, _ = udpCliSock.recvfrom(BUFSIZ)
    if not data:
        break
    print(str(data, encoding="utf-8"))

udpCliSock.close()

File: 4-java_projects/1-Java从入门到放弃/src/com/sxt/udp/UDPClient.java
package com.sxt.udp;

import java.io.IOException;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.net.InetSocketAddress;
import java.net.SocketException;

/**
 * @author: Li Tian
 * @contact: litian_cup@163.com
 * @software: IntelliJ IDEA
 * @file: UDPClient.java
 * @time: 2019/11/14 14:14
 * @desc: 发送端
 */

public class UDPClient {
    public static void main(String[] args) throws IOException {
        System.out.println("发送方启动中...");
        //  1. 使用DatagramSocket指定端口，创建发送端
        DatagramSocket client = new DatagramSocket(8888);
        //  2. 准备数据，一定转成字节数组
        String data = "上海尚学堂";
        byte[] datas = data.getBytes();
        //  3. 封装成DatagramPacket包裹，需要指定目的地
        DatagramPacket packet = new DatagramPacket(datas, 0, datas.length, new InetSocketAddress("localhost", 9999));
        //  4. 发送包裹send(DatagramPacket p)
        client.send(packet);
        //  5. 释放资源
        client.close();
    }
}
